### ISSUE TITLE
How to return the word embeddings and how to understand the hidden_states in return?

### ISSUE THREAD

**Issue by island99** [huggingface/transformers#7114]
# ❓ Questions & Help

## Details

I am new here, transformers is amazing. What I want is to get the embeddings of words with context , “logits” is obviously a bit thin, I think the hidden_states may represent the word embedding, but I can not understand the output of the embeddings and output of each layer.
Which one should I choose?
Thanks very much!

> hidden_states (tuple(torch.FloatTensor), optional, returned when output_hidden_states=True is passed or when config.output_hidden_states=True) – Tuple of torch.FloatTensor (one for the output of the embeddings + one for the output of each layer) of shape (batch_size, sequence_length, hidden_size).

**A link to original question on the forum/Stack Overflow**:

**Comment by LysandreJik** [comment]
Well if you want the embeddings of words with their context, your best bet is to take the entire output of the model. If you take only the word embeddings, these are contextless. 

You should take the model with no head, i.e., the base model: `BertModel`, `GPT2Model`, etc. to get this value.
